### ISSUE TITLE
PHPMAILERSUBSCRIBEHOST does not use smtp username and password

### ISSUE THREAD

**Issue by bramley** [phpList/phplist3#790]
The recent change to use PHPMAILERSUBSCRIBEHOST seems to be incomplete. This bit of code does not set the smtp username and password.
```
        } elseif (!$this->inBlast && empty($_SESSION['adminloggedin']) && 
            defined('PHPMAILERSUBSCRIBEHOST') && 
            defined('PHPMAILERSUBSCRIBEPORT') && PHPMAILERSUBSCRIBEHOST != '') {
#            logEvent('Sending email via PHPMAILERSUBSCRIBEHOST '.PHPMAILERSUBSCRIBEHOST);
            $this->Host = PHPMAILERSUBSCRIBEHOST;
            $this->Port = PHPMAILERSUBSCRIBEPORT;
            $this->Mailer = 'smtp';
```
Adding the username and password will make that chunk of code very similar to the code for PHPMAILERTESTHOST so I wonder what the reason is to have two ways of specifying an alternative smtp server?

**Comment by michield** [comment]
Yes, really the whole thing needs a rewrite. Basically, the idea is that you may want to split out sending hosts for various different purposes:

- transactions (subscribe, confirm, update)
- testing (ie allows sending to test services)
- actually sending campaigns

For each of those you'd need the specific "host, port, user, password, encryption, etc". Not only that, for each of them, you should be able to provide a pool of servers.

So, really, it would be better to have a list of server pools and specify the function of each pool.

This is just a temporary solution. And in fact, I'm still debugging it, so it should not yet be documented or used in production systems.

**Comment by bramley** [comment]
OK but the way that it is coded means that it is not optional. 

PHPMAILERSUBSCRIBEHOST defaults to PHPMAILERHOST and PHPMAILERSUBSCRIBEPORT defaults to PHPMAILERPORT therefore it will be used when someone currently uses smtp, but without providing the smtp credentials.
I think that is what has been reported https://discuss.phplist.org/t/phplist-3-6-4-smtp-code-550/7629

**Comment by michield** [comment]
Ok, yes. A workaround would be to set this in the config

define('PHPMAILERSUBSCRIBEHOST','');

**Comment by michield** [comment]
Resolved with #795
